src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs

```diff
@@ -148,7 +183,8 @@ public static bool VerifySignedZipIntegrity(BinaryReader reader, HashAlgorithm h
                 SignedPackageArchiveIOUtility.HashBytes(hashAlgorithm, BitConverter.GetBytes(eocdrTotalEntries));
                 SignedPackageArchiveIOUtility.HashBytes(hashAlgorithm, BitConverter.GetBytes(eocdrTotalEntriesOnDisk));
 
-                var eocdrSizeOfCentralDirectory = reader.ReadUInt32() - (uint)signatureCentralDirectoryHeader.HeaderSize;
+                // update the central directory size by substracting the length of the signature header size and the central directory header signature
+                var eocdrSizeOfCentralDirectory = (uint)(reader.ReadUInt32() - signatureCentralDirectoryHeader.HeaderSize - BitConverter.GetBytes(SignedPackageArchiveIOUtility.CentralDirectoryHeaderSignature).Length);
```

Comment:
would it be better instead of doing this calculation to have a constant defining the sizes of header signatures? Across all zip headers, the signatures are always 4 bytes.